frappe/public/js/frappe/form/formatters.js

```diff
@@ -53,6 +53,10 @@ frappe.form.formatters = {
 	Currency: function(value, docfield, options, doc) {
 		var currency = frappe.meta.get_field_currency(docfield, doc);
 		var precision = docfield.precision || cint(frappe.boot.sysdefaults.currency_precision) || 2;
+		var temp_value = cstr(value).split(".");
+		if (temp_value.length === 1 || cint(temp_value[1]) === 0) {
+			precision = 2;
+		}
```

Comment:
It does 1.0000 to 1.00

But it does not cover the other scenario like 
1.3000 to 1.3
1.0100 to 1.01
1.2340 to 1.234